### PR TITLE
Update daal4py to 2021.1

### PR DESCRIPTION
**Pull request by Pahandrovich**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -21,14 +24,14 @@ requirements:
     - python
     - setuptools
     - numpy
-    - daal-devel ==2020.3
+    - daal-devel ==2021.1
```
Comment (napetrov) [inline recipe/meta.yaml:27]:
```suggestion
    - dal-devel ==2021.1
```

File: recipe/meta.yaml [hunk 2]
```diff
@@ -21,14 +24,14 @@ requirements:
...
     - cython
     - jinja2
     - mpich      # [not win]
     - clang-tools
   run:
     - python
-    - daal ==2020.3
+    - daal ==2021.1
```
Comment (napetrov) [inline recipe/meta.yaml:34]:
```suggestion
    - dal ==2021.1
```

File: recipe/meta.yaml [hunk 3]
```diff
@@ -53,12 +56,12 @@ test:
     - python -m daal4py examples/sycl/sklearn_sycl.py
 
 about:
-  home: https://intelpython.github.io/daal4py/
+  home: https://intel.github.io/scikit-learn-intelex/
```
Comment (napetrov) [inline recipe/meta.yaml:59]: Don't think so - this is daal4py package and we should point to daal4py documentation

File: recipe/meta.yaml [hunk 4]
```diff
@@ -1,17 +1,20 @@
...
-  url: https://github.com/IntelPython/daal4py/archive/{{ version }}.tar.gz
-  sha256: 5007a706b40242f6f31d5a3bea5a2f38553f1325416df80e38e9c61ac974acf4
+  url: https://github.com/intel/scikit-learn-intelex/archive/{{ version }}.tar.gz
+  sha256: 50f5d274c953b67d419b15c65cb30d9774eea51fb3019fc07321f7e08e3f9274
 build:
   skip: true  # [osx]
   number: {{ buildnumber }}
   include_recipe: false
+  script: |
+    oneapi_env.sh
```
Comment (napetrov) [inline recipe/meta.yaml:16]: we would also need daal4py build with dpcpp on windows

File: recipe/meta.yaml [hunk 5]
```diff
@@ -1,17 +1,20 @@
...
   name: daal4py
   version: {{ version }}
 
 source:
-  url: https://github.com/IntelPython/daal4py/archive/{{ version }}.tar.gz
-  sha256: 5007a706b40242f6f31d5a3bea5a2f38553f1325416df80e38e9c61ac974acf4
+  url: https://github.com/intel/scikit-learn-intelex/archive/{{ version }}.tar.gz
+  sha256: 50f5d274c953b67d419b15c65cb30d9774eea51fb3019fc07321f7e08e3f9274
 build:
   skip: true  # [osx]
```
Comment (napetrov) [inline recipe/meta.yaml:12]:
```suggestion
```

Comment (PetrovKP) [reply recipe/meta.yaml:12]: I propose enable Mac after merge this PR. Because he can works very long

File: recipe/meta.yaml [hunk 6]
```diff
@@ -21,14 +24,14 @@ requirements:
     - python
     - setuptools
     - numpy
-    - daal-devel ==2020.3
+    - dal-devel ==2021.1
```
Comment (napetrov) [inline recipe/meta.yaml:27]:
```suggestion
    - dal-devel ==2021.1.1
```
2021.1.1

File: recipe/meta.yaml [hunk 7]
```diff
@@ -21,14 +24,14 @@ requirements:
...
     - cython
     - jinja2
     - mpich      # [not win]
     - clang-tools
   run:
     - python
-    - daal ==2020.3
+    - dal ==2021.1
```
Comment (napetrov) [inline recipe/meta.yaml:34]:
```suggestion
    - dal ==2021.1.1
```

File: recipe/meta.yaml (hunk 5 above)
Comment (napetrov) [reply recipe/meta.yaml:12]: ok

File: recipe/meta.yaml [hunk 8]
```diff
@@ -1,17 +1,20 @@
-{% set version = "2020.3.1" %}
-{% set buildnumber = 1 %}
+{% set version = "2021.2" %}
```
Comment (napetrov) [inline recipe/meta.yaml:1]: let's go with 2021.1 for now. dal 2021.2 is not yet deployed

File: recipe/meta.yaml [hunk 9]
```diff
@@ -21,14 +24,14 @@ requirements:
     - python
     - setuptools
     - numpy
-    - daal-devel ==2020.3
+    - dal-devel ==2021.1.1
```
Comment (napetrov) [inline recipe/meta.yaml:27]:
```suggestion
    - dal-devel ==2021.1.1
```

**Comment by PetrovKP** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml (hunk 3 above)
Comment (Pahandrovich) [reply recipe/meta.yaml]: Thanks!

**Comment by Pahandrovich** [comment]
@conda-forge-admin, please rerender

**Comment by Pahandrovich** [comment]
@conda-forge-admin, please rerender

**Comment by PetrovKP** [comment]
@conda-forge-admin, please rerender

**Review by PetrovKP — APPROVED** [review]

**Pull request merged**